dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/TrackerUserService.java

```diff
@@ -0,0 +1,72 @@
+package org.hisp.dhis.tracker;
+
+import org.hisp.dhis.common.IdentifiableObjectManager;
+import org.hisp.dhis.hibernate.HibernateUtils;
+import org.hisp.dhis.user.CurrentUserService;
+import org.hisp.dhis.user.User;
+import org.hisp.dhis.user.UserCredentials;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.StringUtils;
+
+/**
+ * Specialized User Service for Tracker that executes User look-up in a read-only transaction
+ *
+ * @author Luciano Fiandesio
+ */
+@Service
+public class TrackerUserService
+{
+    private final CurrentUserService currentUserService;
+
+    private final IdentifiableObjectManager manager;
+
+    public TrackerUserService( CurrentUserService currentUserService, IdentifiableObjectManager manager )
+    {
+        this.currentUserService = currentUserService;
+        this.manager = manager;
+    }
+
+    /**
+     * Fetch a User by user uid
+     * @param userUid a User uid
+     * @return a User
+     */
+    @Transactional( readOnly = true )
+    public User getUser( String userUid )
+    {
+        User user = null;
+
+        if ( !StringUtils.isEmpty( userUid ) )
```

Comment:
Maybe a little bit shaky to have very different behavior based on whether the input argument is null or not in a service method.

Reply:
Yes, I agree. This is the original logic that I extracted from the existing method `getUser`. It goes back to the other message I left in another comment. In general, I don't think that userId should **ever** be null, but this is a bit controversial. I hope that @mortenoh can comment on that,

Reply:
I think the userUid should always be currentUser. but if currentUser == null, this logic wont make a difference, so I don't think we need this logic. currentUser == null when a user did not start the process (IE. if it is a system-started job.) But as we slightly discussed, I dont believe this is a good way to handle it, but I think we can remove this specific if/else logic, since it wont make a difference (getCurrentUser will still return null, if the uid is null)